saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/metadata/OpenSamlMetadataResolver.java

```diff
@@ -163,4 +179,33 @@ private String serialize(EntityDescriptor entityDescriptor) {
 		}
 	}
 
+	public static final class EntityDescriptorParameters {
+
+		private final EntityDescriptor entityDescriptor;
+
+		private final RelyingPartyRegistration registration;
+
+		private final SPSSODescriptor spSsoDescriptor;
```

Comment:
I think this field and the associated method should be left out for the time being. It can be accessed through the `entityDescriptor`.